### ISSUE TITLE
Data format documentation

### ISSUE THREAD

**Issue by almasaeed2010** [LegumeFederation/cmap-js#82]
Hello all,

We recently started using CMap-js (great job btw!), and were wondering about what formats the software accepts. Any information on how to generate data files is also very much appreciated.

Thanks!

**Comment by awilkey** [comment]
Right now the only format fully supported is the old cmap tabular format, which was a choice mainly due to availability and ease of support. After the current round of bug fixes and UI tweaks (with the goal to be able to call it moved out of alpha and into beta) I do have plans on revisiting the data loader and expanding file format support. 

Now is a great time to be requesting additional file formats for support.

**Comment by adf-ncgr** [comment]
agreed, I think it would be good for us (including interested adopters) to spend some time specifying a JSON schema for service provider implementations (chado-backed or otherwise) to target. That said, for use cases involving users reading local data for comparison with maps published on sites we will still want to have a few flatfile options based on software for QTL/GWAS mapping (or similar) that is currently being used by the community. But I agree that FPC is probably not being widely used these days, and could be converted into a cmap format without too much effort (IIRC, the old cmap used to be able to handle FPC maps though maybe there was a conversion step involved).

**Comment by adf-ncgr** [comment]
@sammyjava you will probably want to participate in the discussion of this from the intermine angle too (which must have been in the back of my mind when I wrote: "chado-backed or otherwise")

**Comment by sammyjava** [comment]
I prefer JSON simply because that's what other InterMine displayers use. But otherwise I'm not too particular, I can presumably hack the appropriate file format.

**Comment by jd-campbell** [comment]
FYI @awilkey It came up in the May 24th LegumeFed meeting (short discussion about cmap-js) about the request for a spec for the allowed json format(s).

**Comment by nathanweeks** [comment]
If there's no other widely-used JSON map data format, a JSON representation of the [cmap tab-delimited file format](https://github.com/GMOD/cmap/blob/master/docs/ADMINISTRATION.pod#importing-map-data) *may* at least be straightforward to support in cmap-js.
